web/src/message_view.ts

```diff
@@ -997,6 +997,17 @@ export function maybe_add_local_messages(opts: {
         assert(id_info.final_select_id !== undefined);
 
         if (!load_local_messages(msg_data, superset_data)) {
+            // We don't have the message we want to select locally,
+            // and since our unread data is incomplete, we just
+            // ask server directly for `first_unread`.
+            if (
+                unread.old_unreads_missing &&
+                // Ensure our intent is to narrow to first unread.
+                id_info.final_select_id === unread_info.msg_id &&
+                unread_info.msg_id !== id_info.target_id
+            ) {
+                id_info.final_select_id = undefined;
+            }
```

Comment:
Can you explain why the check is this? I'm not sure why the check isn't on whether `target_id === undefined`...

Reply:
My thought process might be that I wanted to avoid taking user to a different message id if user wants to target that specific unread message id. But, I think `target_id === undefined` is  a better check and simpler to read.